Cython/Compiler/Nodes.py

```diff
@@ -3710,9 +3713,14 @@ def generate_tuple_and_keyword_parsing_code(self, args, success_label, code):
         code.putln("if (%s(%s)) {" % (
             (self.num_required_kw_args > 0) and "likely" or "unlikely",
             Naming.kwds_cname))
-        self.generate_keyword_unpacking_code(
-            min_positional_args, max_positional_args,
-            has_fixed_positional_count, has_kw_only_args, all_args, argtuple_error_label, code)
+
+        if do_generate_kw_unpacking:
+            self.generate_keyword_unpacking_code(
+                min_positional_args, max_positional_args,
+                has_fixed_positional_count, has_kw_only_args, all_args, argtuple_error_label, code)
+        else:
+            code.putln("PyErr_Format(PyExc_TypeError, \"%s() takes no keyword arguments\");" % self.name)
```

Comment:
This looks like a place where we could call `__Pyx_ParseOptionalKeywords()` instead to raise an appropriate error that names the argument. CPython gives me this:
```python
>>> def f(a,b,/): pass
>>> f(a=1,b=2)
Traceback (most recent call last):
TypeError: f() got some positional-only arguments passed as keyword arguments: 'a, b'
```

Reply:
Done